04-onion-routing.md

```diff
@@ -229,39 +239,100 @@ leaking its position in the route.
+      - MUST include `amt_to_forward`, `outgoing_cltv_value` and `total_amount_msat`.
+    - MUST NOT include any other tlv field.
+  - For every node outside of a blinded route:
     - MUST include `amt_to_forward` and `outgoing_cltv_value`.
-  - For every non-final node:
-    - MUST include `short_channel_id`
-    - MUST NOT include `payment_data`
-  - For the final node:
-    - MUST NOT include `short_channel_id`
-    - if the recipient provided `payment_secret`:
-      - MUST include `payment_data`
-      - MUST set `payment_secret` to the one provided
-      - MUST set `total_msat` to the total amount it will send
-    - if the recipient provided `payment_metadata`:
-      - MUST include `payment_metadata` with every HTLC
-      - MUST not apply any limits to the size of payment_metadata except the limits implied by the fixed onion size
+    - For every non-final node:
+      - MUST include `short_channel_id`
+      - MUST NOT include `payment_data`
+    - For the final node:
+      - MUST NOT include `short_channel_id`
+      - if the recipient provided `payment_secret`:
+        - MUST include `payment_data`
+        - MUST set `payment_secret` to the one provided
+        - MUST set `total_msat` to the total amount it will send
+      - if the recipient provided `payment_metadata`:
+        - MUST include `payment_metadata` with every HTLC
+        - MUST not apply any limits to the size of `payment_metadata` except the limits implied by the fixed onion size
 
 The reader:
-  - MUST return an error if `amt_to_forward` or `outgoing_cltv_value` are not present.
-  - if it is not the final node:
-    - MUST return an error if:
-      - `short_channel_id` is not present,
-      - it cannot forward the HTLC to the peer indicated by the channel `short_channel_id`.
-      - incoming `amount_msat` - `fee` < `amt_to_forward` (where `fee` is the advertised fee as described in [BOLT #7](07-routing-gossip.md#htlc-fees))
-      - `cltv_expiry` - `cltv_expiry_delta` < `outgoing_cltv_value`
-  - if it is the final node:
-    - MUST treat `total_msat` as if it were equal to `amt_to_forward` if it
-      is not present.
+
+  - If `encrypted_recipient_data` is present:
+    - If `blinding_point` is set in the incoming `update_add_htlc`:
+      - MUST return an error if `current_blinding_point` is present.
+      - MUST use that `blinding_point` as the blinding point for decryption.
+    - Otherwise:
+      - MUST return an error if `current_blinding_point` is not present.
+      - MUST use that `current_blinding_point` as the blinding point for decryption.
+      - SHOULD add a random delay before returning errors.
+    - MUST return an error if `encrypted_recipient_data` does not decrypt using the
+      blinding point as described in [Route Blinding](#route-blinding).
+    - If `payment_constraints` is present:
+      - MUST return an error if:
+        - the expiry is greater than `encrypted_recipient_data.payment_constraints.max_cltv_expiry`.
```

Comment:
Is a zero `max_cltv_expiry` allowed? 

Bad for privacy, but we don't actually restrict it - further up we say: "It should set `payment_constraints.max_cltv_expiry` to restrict the lifetime of a blinded route"

Reply:
I'm not sure I follow, if you set `max_cltv_expiry` to `0`, payments will always be rejected (since the actual expiry will always be greater than that). If you want an unrestricted lifetime, you would on the contrary set `max_cltv_expiry` to a value far away in the future?

Reply:
I was thinking zero =  unrestricted lifetime but yeah just putting it far in the future makes more sense 👍